header-validator/src/validate-json.ts

```diff
@@ -588,10 +588,29 @@ function clamp<N extends bigint | number>(
   return n
 }
 
-function expiry(ctx: Context, j: Json): Maybe<number | bigint> {
-  return legacyDuration(ctx, j).map((n) =>
-    clamp(ctx, n, limits.sourceExpiryRange[0], limits.sourceExpiryRange[1])
-  )
+function roundAwayFromZeroToNearestDay(n: number): number {
+  if (n <= 0 || !Number.isInteger(n)) {
+    throw new RangeError()
+  }
+
+  n += secondsPerDay / 2
```

Comment:
nit. we could use a temporary variable instead of mutating the function argument.

Reply:
Done.